src/sagemaker_containers/_intermediate_output.py

```diff
@@ -97,7 +97,9 @@ def _watch(inotify, watchers, watch_flags, s3_uploader):
                 # for it which should cause any problems because when we copy files to temp dir
                 # we add a unique timestamp up to microseconds.
                 if flag is inotify_simple.flags.ISDIR and inotify_simple.flags.CREATE & event.mask:
-                    for folder, dirs, files in os.walk(os.path.join(intermediate_path, event.name)):
+                    path = os.path.join(
+                    *[intermediate_path] + [watchers[wd] for wd in range(1, event.wd + 1)] + [event.name])
```

Comment:
Please extract this logic outside `os.path.join` to be more readable and easier to understand.

Reply:
There was an issue with this. I have fixed it in the latest commit.